audiomentations/core/composition.py

```diff
@@ -259,31 +259,45 @@ def __call__(self, *args, **kwargs):
 class OneOf(BaseCompose):
     """
     OneOf randomly picks one of the given transforms when called, and applies that
-    transform.
+    transform. Optional `weights` can be supplied to guide the probability of each transform being chosen.
     Usage example:
     ```
     augment = OneOf([
         TimeStretch(min_rate=0.8, max_rate=1.25, p=1.0),
         PitchShift(min_semitones=-4, max_semitones=4, p=1.0),
-    ])
+    ], weights=[0.8, 0.2])
     # Generate 2 seconds of dummy audio for the sake of example
     samples = np.random.uniform(low=-0.2, high=0.2, size=(32000,)).astype(np.float32)
     # Augment/transform/perturb the audio data
     augmented_samples = augment(samples=samples, sample_rate=16000)
-    # Result: The audio was either time-stretched or pitch-shifted, but not both
+    # Result: The audio was either time-stretched (less likely) or pitch-shifted (4x more likely), but not both
     ```
     """
 
-    def __init__(self, transforms, p: float = 1.0):
+    def __init__(self, transforms, p: float = 1.0, weights: list[float] | None = None):
         super().__init__(transforms, p)
         self.transform_index = 0
         self.should_apply = True
+        
+        # Handle weights initialization and validation
+        if weights is not None:
+            if len(weights) != len(transforms):
+                raise ValueError("Length of weights must match length of transforms")
+            if any(w < 0 for w in weights):
+                raise ValueError("All weights must be non-negative")
+            if sum(weights) == 0:
+                raise ValueError("Sum of weights must be greater than 0")
+            # Normalize weights to sum to 1
+            self.weights = np.array(weights) / sum(weights)
```

Comment:
This should be faster, given that the sum is calculated with numpy instead of python

```suggestion
            self.weights = np.array(weights)
            self.weights /= np.sum(self.weights)
```